src/Fixer/StringNotation/HeredocClosingMarkerFixer.php

```diff
@@ -0,0 +1,174 @@
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\StringNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Michael Vorisek <https://github.com/mvorisek>
+ */
+final class HeredocClosingMarkerFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Unify `heredoc` or `nowdoc` closing marker.',
+            [
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<"TEST"
+                        Foo
+                        TEST;
+
+                        EOD
+                ),
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<'TEST'
+                        Foo
+                        TEST;
+
+                        EOD,
+                    ['closing_marker' => 'EOF']
+                ),
+                new CodeSample(
+                    <<<'EOD'
+                        <?php $a = <<<TEST
+                        Foo
+                        TEST;
+
+                        EOD,
+                    ['explicit_heredoc_style' => true]
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_START_HEREDOC);
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder(
+                'closing_marker',
+                'Preferred closing marker.'
+            ))
+                ->setAllowedTypes(['string'])
+                ->setDefault('EOD')
+                ->getOption(),
+            (new FixerOptionBuilder(
+                'reserved_closing_markers',
+                'Reserved closing markers to be kept unchanged.'
+            ))
+                ->setAllowedTypes(['array'])
+                ->setDefault([
+                    'CSS',
+                    'DIFF',
+                    'HTML',
+                    'JS',
+                    'JSON',
+                    'MD',
+                    'PHP',
+                    'PYTHON',
+                    'RST',
+                    'TS',
+                    'SQL',
+                    'XML',
+                    'YAML',
+                ])
```

Comment:
If there is no API to extract option default value from a rule, yes, it should be added, but not related with this PR :)

Reply:
I am closing this thread as resolved, feel free to create feature request from it.

Reply:
There's no way to get defaults from the config or extend them instead of overriding. I would like to see this in introducing PR, because it's a minute of work on existing branch (new PR requires more work and other maintainer's approval).

Reply:
quite atypical, searched `public const.*\[`, but done